lib/tty/pager/system.rb

```diff
@@ -123,22 +124,37 @@ def initialize(**options)
       #
       # @api public
       def page(text, &_callback)
-        return text unless output.tty?
+        write(text)
+        wait
+      end
+
+      def start
+        # In case there's a previous pager running:
+        wait
 
         command = pager_command
         out = self.class.run_command(command)
         # Issue running command, e.g. unsupported flag, fallback to just command
-        if !out.empty?
+        unless out.empty?
           command = pager_command.split.first
         end
 
-        pager_io = open("|#{command}", 'w')
```

Comment:
The intention behind `tty` gems is to work, if possible, on all systems including Windows. The reason why the `|` pipe is used in this instance, is to make the process piping work on Windows. Have you checked your change against Windows?

Reply:
Hm, I admit, I didn't test it on Windows. I didn't know this was the reason. Switching to `IO.popen` had the nice qualities of silencing the Hound warning and making it much more reliable to stub (though I could probably make stubbing just as easy by using `Kernel.open` directly rather than through the class). I'll see what I can do to test it out.

Reply:
I'm all for switching over to `popen` if it makes things nicer but would like to keep the windows support as well. Maybe we can have both with this. I just don't know without testing. I picked up this particular 'trick' from Avdi Grimm and it reduced my 'hacky' code to this wonderful one character.

Reply:
Okay, so I tested it on Windows 10, and it seems to work just fine:
<img width="960" alt="tty-prompt" src="https://user-images.githubusercontent.com/124255/80312378-9846d800-87ed-11ea-8f7c-b333e634c7d9.png">

I'm not 100% sure if this will work on all setups, since ruby and git install a bunch of development stuff, but I also feel like `Kernel.open` probably just delegates to this anyway? The problem with it is it's also used to open files and URIs, so `popen` is clearer in terms of intent, if nothing else. I'd keep it for now.

Reply:
That looks promising. Thanks for checking Windows. I normally test on XP VM - a bit old school I know. I don't know the internals of `popen` but semantically it seems to be doing what the pipe character used to do - open a child process and pipe to it. Let's go with `popen` for now.